hardpy/pytest_hardpy/db/json_file_store.py

```diff
@@ -0,0 +1,169 @@
+                temp_file.unlink()
+            raise
+
+    def update_doc(self) -> None:
+        """Reload document from JSON file to memory."""
+        if self._file_path.exists():
+            try:
+                with self._file_path.open("r") as f:
+                    self._doc = json.load(f)
+            except json.JSONDecodeError as exc:
+                self._log.error(f"Error reading storage file: {exc}")
+                # Keep existing in-memory document if file is corrupted
+            except Exception as exc:
+                self._log.error(f"Error reading storage file: {exc}")
+                raise
+
+    def get_document(self) -> BaseModel:
+        """Get full document with schema validation.
+
+        Returns:
+            BaseModel: Validated document model
+        """
+        self.update_doc()
+        return self._schema(**self._doc)
+
+    def clear(self) -> None:
+        """Clear storage by resetting to initial state (in-memory only).
+
+        Note: For JSON storage, clear() only resets the in-memory document.
+        Call update_db() explicitly to persist the cleared state.
+        This differs from CouchDB where clear() immediately affects the database.
+        """
+        # Reset document to initial state (in-memory only)
+        self._doc = self._create_default_doc_structure(self._doc_id)
+        # NOTE: We do NOT call update_db() here to avoid persisting cleared state
+        # The caller should call update_db() when they want to persist changes
+
+    def compact(self) -> None:
+        """Optimize storage (no-op for JSON file storage)."""
+
+    def _init_doc(self) -> dict:
+        """Initialize or load document structure.
+
+        Returns:
+            dict: Document structure
+        """
+        if self._file_path.exists():
+            try:
+                with self._file_path.open("r") as f:
+                    doc = json.load(f)
+
+                    # Ensure required fields exist (for backward compatibility)
+                    if DF.MODULES not in doc:
+                        doc[DF.MODULES] = {}
+
+                    # Reset volatile fields for state-like stores
+                    if self._store_name == "statestore":
+                        default_doc = self._create_default_doc_structure(doc["_id"])
+                        doc[DF.DUT] = default_doc[DF.DUT]
+                        doc[DF.TEST_STAND] = default_doc[DF.TEST_STAND]
+                        doc[DF.PROCESS] = default_doc[DF.PROCESS]
+
+                    return doc
+            except json.JSONDecodeError:
+                self._log.warning(
+                    f"Corrupted storage file {self._file_path}, creating new",
+                )
+            except Exception as exc:  # noqa: BLE001
+                self._log.warning(f"Error loading storage file: {exc}, creating new")
+
+        # Return default document structure
+        return self._create_default_doc_structure(self._doc_id)
```

Comment:
In order to maintain compatibility with the schema validator, the `_rev` field must be added. To verify, you can run the plugin tests by specifying `storage_type = “json”` in the [hardpy.toml](https://github.com/rici4kubicek/hardpy/blob/feature/json_storage/tests/test_plugin/hardpy.toml) file.

```suggestion
        return self._create_default_doc_structure(self._doc_id)

    def _create_default_doc_structure(self, doc_id: str) -> dict:
        # CouchDB compatibility
        doc = super()._create_default_doc_structure(doc_id)
        doc["_rev"] = self._doc_id
        return doc
```